db/migrate/20150612135546_create_host_status.rb

```diff
@@ -33,4 +40,41 @@ def down
 
     drop_table :host_status
   end
+
+  private
+
+  def update_statuses(host)
+    results = HostStatus.status_registry.map do |status_class|
+      status = host.get_status(status_class)
+      update_sub_status(status)
+    end
+
+    if results.any?
+      host.host_statuses.reload
+      host.refresh_global_status
+
+      unless (saved = host.save(:validate => false))
+        logger.warn "Skipping global status refresh for host #{host.name}"
+      end
+      results.push saved
+    end
+
+    !results.include?(false)
```

Comment:
As @iNecas says, this will only cause warning, if users are curious they are encouraged to look at logs, we don't want to print thousands of logs if thousands host statuses fails. Since statuses will get updated automatically later, no action should be needed.